services/theme_update.py

```diff
@@ -205,7 +207,7 @@ def application(environ, start_response):
 
     status = '200 OK'
     with statsd.timer('services.theme_update'):
-        data = environ['wsgi.input'].read()
+        environ['wsgi.input'].read()
```

Comment:
This had me shocked tbh… This does not take content-length into account at all and just reads the whole bloody input stream… This should somehow use `django.core.handlers.wsgi:LimitedStream` but on the other hand… why on earth are we evaluating `wsgi.input` anyway, afaik this is `GET` only right? If so then imho `wsgi.input` should not matter to us. See http://wsgi.tutorial.codepoint.net/parsing-the-request-post for a bit more background information.

Reply:
Or use `environ.get('CONTENT_LENGTH', 0)` for the content-length and default to something permissive (like 2MB).

Reply:
Could be that all this is less of an problem since we're protected via nginx. But tbh, I have absolutely no clue, there's a very good reason I am not writing any raw wsgi code since 2010 anymore 😆

Reply:
yeah I've no idea what it was doing - the pattern didn't make much sense to me, so i just removed the unused variable to quiet the warning.  I'll take a proper look 😉

Reply:
gone https://github.com/mozilla/addons-server/pull/6583/commits/5bc410239df486032d8cf27a7bd465747fe57e88